hudi-common/src/main/java/org/apache/hudi/common/util/ParquetUtils.java

```diff
@@ -105,6 +122,60 @@
     return rowKeys;
   }
 
+  /**
+   * Read the rows with record key and partition path from the given parquet file.
+   *
+   * @param filePath      The parquet file path.
+   * @param configuration configuration to build fs object
+   * @return Set Set of row keys matching candidateRecordKeys
+   */
+  public static List<Pair<HoodieKey, HoodieRecordLocation>> fetchRecordKeyPartitionPathFromParquet(Configuration configuration, Path filePath,
+                                                                                                   String baseInstantTime,
+                                                                                                   String fileId) {
+    return fetchRecordKeyPartitionPathFromParquet(configuration, filePath, baseInstantTime, fileId, null);
+  }
+
+  /**
+   * Read the rows with record key and partition path from the given parquet file.
+   *
+   * @param filePath      The parquet file path.
+   * @param configuration configuration to build fs object
+   * @return Set Set of row keys matching candidateRecordKeys
+   */
+  static List<Pair<HoodieKey, HoodieRecordLocation>> fetchRecordKeyPartitionPathFromParquet(Configuration configuration, Path filePath,
+                                                                                            String baseInstantTime,
+                                                                                            String fileId, List<String> recordsToFilter) {
+    List<Pair<HoodieKey, HoodieRecordLocation>> rows = new ArrayList<>();
+    try {
+      if (!filePath.getFileSystem(configuration).exists(filePath)) {
+        return new ArrayList<>();
+      }
+
+      Configuration conf = new Configuration(configuration);
+      conf.addResource(FSUtils.getFs(filePath.toString(), conf).getConf());
+      Schema readSchema = HoodieAvroUtils.getRecordKeyPartitionPathSchema();
+      AvroReadSupport.setAvroReadSchema(conf, readSchema);
+      AvroReadSupport.setRequestedProjection(conf, readSchema);
+      ParquetReader reader = AvroParquetReader.builder(filePath).withConf(conf).build();
+      Object obj = reader.read();
+      while (obj != null) {
+        if (obj instanceof GenericRecord) {
+          String recordKey = ((GenericRecord) obj).get(HoodieRecord.RECORD_KEY_METADATA_FIELD).toString();
+          String partitionPath = ((GenericRecord) obj).get(HoodieRecord.PARTITION_PATH_METADATA_FIELD).toString();
+          if (recordsToFilter == null) {
+            rows.add(Pair.of(new HoodieKey(recordKey, partitionPath), new HoodieRecordLocation(baseInstantTime, fileId)));
```

Comment:
can we just limit this method to reading the parquet fields and nothing else.. i.e you can add the `HoodieRecordLocation` object outside on the caller side.